bin/init.bash

```diff
@@ -8,44 +8,52 @@
 set -eu -o pipefail
 shopt -s globstar nullglob dotglob
 
-if [ $# -lt 1 ] || [ $# -gt 2 ]; then
-    echo "error when running $0: argument mismatch" 1>&2
-    exit 1
-fi
-
-flavor=$1
-if [ $# -eq 2 ]; then
-    fingerprint=$2
-fi
-
 here="$(dirname "$(readlink -f "$0")")"
 # shellcheck source=bin/common.bash
 source "${here}/common.bash"
 
-CK8S_CLOUD_PROVIDER=${CK8S_CLOUD_PROVIDER:-""}
-if [[ ${CK8S_CLOUD_PROVIDER} != "" ]]; then
-    log_error "ERROR: CK8S_CLOUD_PROVIDER is not supported"
+cloud_provider="${CK8S_CLOUD_PROVIDER:-}"
+fingerprint="${CK8S_PGP_FP:-}"
+
+while [ "${#}" -gt 1 ]; do
+    case "${1}" in
+    "--cloud-provider") cloud_provider="${2}" ;;
+    "--sops-fingerprint") fingerprint="${2}" ;;
+    *)
+        log_error "ERROR: unknown flag ${1}"
+        exit 1
+        ;;
+    esac
+    shift 2
+done
+
+if [ -z "${cloud_provider}" ]; then
+    log_error "ERROR: either --cloud-provider or the environment variable CK8S_CLOUD_PROVIDER must be set."
     exit 1
 fi
 
-# Validate the flavor
-if [ "${flavor}" != "default" ] && \
-   [ "${flavor}" != "gcp" ] && \
-   [ "${flavor}" != "openstack" ] && \
-   [ "${flavor}" != "vsphere" ] && \
-   [ "${flavor}" != "aws" ]; then
-    log_error "ERROR: Unsupported flavor: ${flavor}"
-    exit 1
+config_type="default"
+if [ -n "${cloud_provider:-}" ]; then
+    case "${cloud_provider:-}" in
+        aws|gcp|vsphere)
+            config_type="${cloud_provider}"
+            ;;
+
+        citycloud|safespring)
+            config_type="openstack"
+            ;;
+
+        *)
+            log_error "ERROR: Unsupported cloud provider: ${cloud_provider}"
+            exit 1
+            ;;
+    esac
```

Comment:
This does not support exoscale. Should there be a way to generate a "generic" config as well?
```suggestion
if [ -n "${cloud_provider:-}" ]; then
    case "${cloud_provider:-}" in
        aws|gcp|vsphere)
            config_type="${cloud_provider}"
            ;;

        citycloud|safespring)
            config_type="openstack"
            ;;
            
        exoscale)
            ;;

        *)
            log_error "ERROR: Unsupported cloud provider: ${cloud_provider}"
            exit 1
            ;;
    esac
```